### PR TITLE
34 uploading file

### PR DESCRIPTION
**Pull request by hamedveloper**
# Description

This story is about a function that when invoked with a file as a argument will upload the file. It returns a url for accessing that file! 

Fixes https://github.com/HackYourFuture-CPH/rate-my-cv/issues/34

# How to test?

For making a test you should first creat a project in firebase and a database in there as well. You should change the firebase Config here in the code based on the configure parameters in firebase. Afterward give a file to the function and in the result you can give a Url of that updated file. 
 
# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/hooks/fileUploader.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { storage } from '../firebase/configure';
+
+function FileUploader(cv) {
+  cv.preventDefault();
+
+  const uploadTask = storage.ref(`/CVs/${cv.name}`).put(cv);
+  uploadTask.on(
+    'state_changed',
+    (snapShot) => {
+      console.log(snapShot);
```
Comment (benna100) [inline src/client/hooks/fileUploader.js:10]: remove the console.log

File: src/client/hooks/fileUploader.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { storage } from '../firebase/configure';
+
+function FileUploader(cv) {
+  cv.preventDefault();
+
+  const uploadTask = storage.ref(`/CVs/${cv.name}`).put(cv);
```
Comment (benna100) [inline src/client/hooks/fileUploader.js:6]:
```suggestion
  const uploadTask = storage.ref(`/cvs/${cv.name}`).put(cv);
```

**Review by benna100 — CHANGES_REQUESTED** [review]
the changes in package.json is not part of the story

**Comment by benna100** [comment]
@hamedveloper are you still working on this?

**Comment by hamedveloper** [comment]
Yes I am. I just finished with another story and comming back to finish
this one today

**Comment by hamedveloper** [comment]
@benna100 Please check. I have done it in another way. 


https://user-images.githubusercontent.com/54910388/106212978-09c41480-61cc-11eb-931a-5f5471ae53cb.mp4

**Review by benna100 — DISMISSED** [review]
Hamed you need to clean this PR up. There are lots and lots of changes not related to your story

**Comment by hamedveloper** [comment]
Can you let me know how I can clean up changes? It did not let me commit because of formatting and as soon as I made other files format let me to commit. and now I can see I have conflict here!!

**Comment by pdsorensen** [comment]
@hamedveloper, if you go into the "files changed" section of your PR, you can see alot of changes has been made to files that it shouldn't be changed. You also need to merge this branch with the latest changes in develop. 

Cheers,

File: src/client/hooks/fileUploader.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          const percentage = Math.round(
+            (snap.bytesTransferred / snap.totalBytes) * 100,
+          );
+          setProgress(percentage);
+        },
+        (err) => {
+          setError(err);
+        },
+        async () => {
+          const downloadUrl = await storageRef.getDownloadURL();
```
Comment (valmdz) [inline src/client/hooks/fileUploader.js:25]:
```suggestion
          const downloadedUrl = await storageRef.getDownloadURL();
```

File: src/client/hooks/fileUploader.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          // save the url to local state
+          setUrl(downloadUrl);
```
Comment (valmdz) [inline src/client/hooks/fileUploader.js:27]:
```suggestion
          setUrl(downloadedUrl);
```

**Review by pdsorensen — APPROVED** [review]
@hamedveloper this is great! Great video as well, makes it really easy to see what you have done. Great job also trimming the PR down. 💯

**Pull request merged**